src/widgets/Tabs/VaccineSelect.js

```diff
@@ -147,19 +165,65 @@ const VaccineSelectComponent = ({
 
       <FlexRow flex={1} alignItems="flex-end" justifyContent="flex-end">
         <PageButtonWithOnePress text={buttonStrings.cancel} onPress={onCancelPrescription} />
-        <PageButtonWithOnePress
-          text={buttonStrings.confirm}
-          style={{ marginLeft: 'auto' }}
-          isDisabled={!selectedBatches && !hasRefused}
-          onPress={confirmPrescription}
+
+        {wasVaccinatedToday() ? (
+          <PageButton
+            text={buttonStrings.confirm}
+            style={{ marginLeft: 'auto' }}
+            isDisabled={!selectedBatches && !hasRefused}
+            onPress={toggleConfirmDoubleDoseModal}
+          />
+        ) : (
+          <PageButtonWithOnePress
+            text={buttonStrings.confirm}
+            style={{ marginLeft: 'auto' }}
+            isDisabled={!selectedBatches && !hasRefused}
+            onPress={confirmPrescription}
+          />
+        )}
+
+        {wasVaccinatedToday() ? (
+          <PageButton
+            text={generalStrings.ok_and_next}
+            style={{ marginLeft: 5 }}
+            isDisabled={!selectedBatches && !hasRefused}
+            onPress={toggleConfirmAndRepeatDoubleDoseModal}
+          />
+        ) : (
+          <PageButtonWithOnePress
+            text={generalStrings.ok_and_next}
+            style={{ marginLeft: 5 }}
+            isDisabled={!selectedBatches && !hasRefused}
+            onPress={confirmAndRepeatPrescription}
+          />
+        )}
```

Comment:
same suggestion as above.. as long as we prevent the accidental double click on the buttons we can just use `PageButton`